### PR TITLE
VIVO-1961: GitHub Action to Publish Docker Image

### PR DESCRIPTION
**Pull request by wwelling**
Resolves [VIVO-1961](https://jira.lyrasis.org/browse/VIVO-1961)

# What does this pull request do?

Simple GitHub action workflow and Dockerfile to build VIVO tomcat image and publish to [vivoweb/vivo](https://hub.docker.com/r/vivoweb/vivo)

# What's new?

`Dockerfile` and `.github/workflows/deploy.yml`

# How should this be tested?

The Dockerfile can be tested locally following these commands with Docker installed.

```
mvn clean install
docker build --no-cache -t vivo .
docker run -p 8080:8080 vivo
```

Or using the published image with a single command.

```
docker run -p 8080:8080 vivoweb/vivo:latest
```

Then navigate the browser to http://localhost:8080. You should see VIVO running as root Tomcat webapp. It will not have search functionality.

Solr and database sold separately.

@VIVO-project/vivo-committers

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Maven Build
+        run: mvn clean install
+
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          repository: vivoweb/vivo
+          tag_with_ref: true
```
[the remark this answers is not in the document]
Comment (wwelling) [reply .github/workflows/docker.yml:43]: Latest makes sense for the snapshot images. I also want to upgrade to the latest version of build-push-action.

[the remark this answers is not in the document]
Comment (wwelling) [reply .github/workflows/docker.yml:43]: Also, will update how to test using the published image.

**Review by awoods — APPROVED** [review]
Starts right up:
```
docker pull vivoweb/vivo:VIVO-1961
docker run -p8080:8080 vivoweb/vivo:VIVO-1961
```

I am not connecting to my local Solr, which is not surprising since Docker's "localhost" is not my laptop.
This is a great start.

**Comment by awoods** [comment]
We will likely want to provide a "docker-compose.yml" that brings up a Solr instance and networks the two instances together. The trick will be to configure the Solr schema in the process.

**Comment by wwelling** [comment]
Andrew, I will gladly include a Solr Dockerfile and config with docker-compose.yml to spin up fully functional VIVO in this PR. Would also like for the docker-compose.yml to include a volume for VIVO home.

**Comment by awoods** [comment]
For the sake of other reviewers, my inclination is to keep PRs focused and small, moving this PR forward without additional scope... followed by iteration: docker-compose.yml, configurable runtime.properties, configured Solr, etc.

But maybe that is an ideal flow where the committers have bandwidth for quick reviews. In the absence of that, I don't want to stall your fantastic progress.

**Comment by wwelling** [comment]
Works for me. I will leave it here with an upgrade of the community action for publishing docker images.

**Review by awoods — APPROVED** [review]

**Pull request merged**